gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRTaskStateTracker.java

```diff
@@ -117,4 +113,40 @@ protected void updateTaskMetrics() {
       this.context.progress();
     }
   }
+  
+  private void updateCounters(Task task) {
+    updateCounters(task, MetricGroupFilter.JOB);
```

Comment:
updateCounters() is outside MRTaskMetricsUpdater, there no reference to this.task.